### PR TITLE
Riak driver

### PR DESCRIPTION
**Pull request by brenopessoa**
Please check this initial proposal for Riak Driver implementation.

### CODE REVIEW

File: riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManager.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+			throws NullPointerException, UnsupportedOperationException {
+
+		K key = entity.getKey();
+		Value value = entity.getValue();
+
+		StoreValue storeValue = createStoreValue(key,value,nameSpace,ttl);
+		
+		try {
+			client.execute(storeValue);
+		} catch (ExecutionException e) {
```
Comment (otaviojava) [inline riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManager.java]: why Not a mult-try?

```java
} catch (ExecutionException | InterruptedException e) {
throw new DianaRiakException(e.getMessage(),e);
}
```

File: riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManager.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+		FetchValue fetchValue = createFetchValue(nameSpace, key);
+		
+		try {
+			FetchValue.Response response = client.execute(fetchValue);
+			String valueFetch = response.getValue(String.class);
+			if(StringUtils.isNoneBlank(valueFetch))
+				 return Optional.of(RiakValue.of(gson,valueFetch));
+	    
+		} catch (UnresolvedConflictException e) {
```
Comment (otaviojava) [inline riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManager.java]: You can improve these exceptions with multy-try exception from Java 7

File: riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManager.java [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+				Response response = client.execute(fetchValue);
+				
+				if(response.hasValues()){
+					
+					String value = response.getValue(String.class);
+					if(StringUtils.isNoneBlank(value)){
+						values.add(RiakValue.of(gson,value));
+					}
+				}
+			} catch (ExecutionException e) {
```
Comment (otaviojava) [inline riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManager.java]: multi-try exception.

File: riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManager.java [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+			throw new DianaRiakException(e.getMessage(),e);
+		}
+	     
+		return Optional.empty();
+	}
+
+	@Override
+	public <K> Iterable<Value> get(Iterable<K> keys) throws NullPointerException {
+		
+	/* Stream Version , is so ugly , but i waiting feedback for implemententions
```
Comment (otaviojava) [inline riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManager.java]: death code

File: riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManager.java [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+	}
+
+	@Override
+	public <K> void remove(K key) throws NullPointerException {
+
+		DeleteValue deleteValue = createDeleteValue(nameSpace,key);
+		
+		try {
+			client.execute(deleteValue);
+		} catch (ExecutionException e) {
```
Comment (otaviojava) [inline riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManager.java]: multi-try exception

File: riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManagerFactory.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+		
+		cluster.start();
+		RiakClient riakClient = new RiakClient(cluster);
+		Namespace quotesBucket = new Namespace(bucketName);
+
+		return new RiakKeyValueEntityManager(riakClient, new Gson(),quotesBucket);
+	}
+
+	@Override
+	public <T> List<T> getList(String bucketName, Class<T> clazz) throws UnsupportedOperationException {
```
Comment (otaviojava) [inline riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManagerFactory.java]: Does Riak can support this method? 
If not, just throw an exception: 
`throw new UnsupportedOperationException("There is not support to getList on Riack");`

File: riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManagerFactory.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public <T> Set<T> getSet(String bucketName, Class<T> clazz) throws UnsupportedOperationException {
```
Comment (otaviojava) [inline riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManagerFactory.java]: Same problem here.

File: riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManagerFactory.java [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public <T> Queue<T> getQueue(String bucketName, Class<T> clazz) throws UnsupportedOperationException {
```
Comment (otaviojava) [inline riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManagerFactory.java]: Same problem here.

File: riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManagerFactory.java [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+import org.jnosql.diana.api.key.BucketManagerFactory;
+
+import com.basho.riak.client.api.RiakClient;
+import com.basho.riak.client.core.RiakCluster;
+import com.basho.riak.client.core.query.Namespace;
+import com.google.gson.Gson;
+
+public class RiakKeyValueEntityManagerFactory implements BucketManagerFactory<RiakKeyValueEntityManager> {
+
+	RiakCluster cluster;
```
Comment (otaviojava) [inline riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManagerFactory.java]: why not that's a `private`

File: riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManager.java (hunk 4 above)
Comment (brenopessoa) [inline riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManager.java]: What version of implemetation you prefer?

https://gist.github.com/brenopessoa/7aaceaa56dfeefd41a8f1754e0ca6856

or

https://gist.github.com/brenopessoa/e42d0642d615e2607ba1813d6c43bb23

File: riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManager.java (hunk 1 above)
Comment (brenopessoa) [inline riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManager.java]: I will fix it.

File: riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManager.java (hunk 2 above)
Comment (brenopessoa) [inline riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManager.java]: I will fix it.

File: riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManager.java (hunk 3 above)
Comment (brenopessoa) [inline riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManager.java]: I will fix it.

File: riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManager.java (hunk 5 above)
Comment (brenopessoa) [inline riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManager.java]: I will fix it.

File: riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManagerFactory.java (hunk 6 above)
Comment (brenopessoa) [inline riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManagerFactory.java]: I will  verify the support  implementation of this methods or throw an exception.

File: riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManagerFactory.java (hunk 7 above)
Comment (brenopessoa) [inline riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManagerFactory.java]: I will  verify the support  implementation of this methods or throw an exception.

File: riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManagerFactory.java (hunk 8 above)
Comment (brenopessoa) [inline riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManagerFactory.java]: I will  verify the support  implementation of this methods or throw an exception.

**Pull request merged**